lib/core/src/recover/recoverer.rs

```diff
@@ -161,11 +163,20 @@ impl Recoverer {
                         log::warn!("Could not apply recovered data for Send swap {swap_id}: recovery data not found");
                         continue;
                     };
-                    let is_expired = liquid_height
-                        >= send_swap.get_boltz_create_response()?.timeout_block_height as u32;
+                    let timeout_block_height =
+                        send_swap.get_boltz_create_response()?.timeout_block_height as u32;
+                    let is_expired = liquid_tip.height >= timeout_block_height;
                     if let Some(new_state) = recovered_data.derive_partial_state(is_expired) {
                         send_swap.state = new_state;
                     }
+                    // Stop updating the expiry when the swap is expired or complete
+                    if !is_expired && send_swap.state.is_ongoing() {
```

Comment:
Is there a reason not to set the expiry even for not in progress payments? I think it is easier to think of this field as always set.

Reply:
It was to avoid having to collect historic block headers, as the header time is used to calculate the expiry. 

We can add this. If the current tip is after the timeout block height, we should fetch that block header and set the expiry to that time.

Reply:
Addressed in 283ad39c21b0